src/plasma-cash/entity.ts

```diff
@@ -234,20 +234,22 @@ export class Entity {
 
   async respondChallengeBeforeAsync(params: {
     slot: BN
-    challengingBlockNum: BN
+    challengingTxHash: string
+    respondingBlockNum: BN
   }): Promise<object> {
-    const { slot, challengingBlockNum } = params
-    const challengingBlock = await this._dAppPlasmaClient.getPlasmaBlockAtAsync(
-      challengingBlockNum
+    const { slot, challengingTxHash, respondingBlockNum } = params
+    const respondingBlock = await this._dAppPlasmaClient.getPlasmaBlockAtAsync(
+      respondingBlockNum
     )
-    const challengingTx = await challengingBlock.findTxWithSlot(slot)
-    if (!challengingTx) {
-      throw new Error(`Invalid challenging block: missing tx for slot ${slot.toString(10)}.`)
+    const respondingTx = await respondingBlock.findTxWithSlot(slot)
```

Comment:
I think we use to add a suffix of `Async` on functions with `async/await`, like `findTxWithSlotAsync`

Reply:
Ah, you found a bug, `findTxWithSlot` is not async, my bad. @gakonst can you please remove `await` from all calls to `findTxWithSlot` in this file?